### DISCUSSION TITLE
Danfoss Ally (Popp, Hive) thermostats occupied heating setpoint

### DISCUSSION THREAD

**Discussion by Retr0Joe** [Koenkk/zigbee2mqtt#20910] [Devices]
Hello. I have a "problem" with zigbee2mqtt and danfoss ally room thermostats 014G2461. The thermostats support two ways to define the target temperature. 

1st option is "occupied_heating_setpoint" which sets the thermostat to the defined temperature and the thermostats should react normally fast. Some people thought, that this can lead to overshoot temperature in a room, but that is highly depending on house and rooms, temperaure losses..

2nd option is "occupied_heating_setpoint_scheduled" which lets the room temperature slowly/smooth react to temperature alterations.

The actual problem is, then when you set climate set temperature from home assistant or locally change at the thermostat, every time "occupied_heating_setpoint_scheduled" will be called. From my understanding, this is not how the thermostat is designed to work. When you look at the device page, every time when you do a temperature change, both values "occupied_heating_setpoint" and "occupied_heating_setpoint_scheduled" will be changed, which cannot be right and could lead to weird behaviours. Only 1 option should be sent to the devices, never both.

I think the solution could be:

- When temperature is set locally at the thermostat, only "occupied heating setpoint" should be called.

- When "Algorithm scale factor" is set to 0, then "occupied heating setpoint" should be called for remote temperature control.
This needs to be added, actually only values between 1 and 10 are possible.

- When "Algorithm scale factor" is between 1 and 10, then "occupied heating setpoint scheduled" should be called for remote temperature control.

@koenkk
I am not sure, where or what needs to be changed. Is it possible, that we change this for testing? I could test these changes and report back then. You can also see, that the thermostats report "Setpoint change source" back, when you change the temperature remotely, localy (maualy) or scheduled,

When you look at forums, many people have problems with the temperature control of these devices. I added some photos where you see, how slow these valves are reacting. The "learning" adapation run was finished, that isnt the problem.

I found many topics, where some people had nearly similar problems. At once, see:

https://github.com/Koenkk/zigbee2mqtt/discussions/12606
https://github.com/Koenkk/zigbee2mqtt/issues/15256
https://github.com/Koenkk/zigbee2mqtt/discussions/19495
https://github.com/Koenkk/zigbee-herdsman-converters/pull/4734

**Comment by Retr0Joe** [comment]
![1](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/8072da89-8524-4334-ae95-ad22b2a5c870)

![2](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/9de6a5db-18c4-48e0-b75c-f42f442a2040)

[AM375549618098en-000103.pdf](https://github.com/Koenkk/zigbee2mqtt/files/13997675/AM375549618098en-000103.pdf)

[AU417130778872en-000101.pdf](https://github.com/Koenkk/zigbee2mqtt/files/13997676/AU417130778872en-000101.pdf)

**Comment by robertalexa** [comment]
My 2 cents on this, whatever the outcome of this will be, please take into consideration the first paragraph of my very old comment https://github.com/Koenkk/zigbee-herdsman-converters/pull/4734#issuecomment-1321170691

The slow valve opening used to be the default in multiple implementations, and everyone hated it. Just dont change it so that the default is slow again. If algorithm factor gets taken into account for what command should be called, remember to document the behaviour and correlation.

Also happy to contribute to testing should time allow me. I still have the hive trvs as daily drivers.

**Reply by surrealo** [reply]
hmm, I'm confused, I thought  you could choose what you want to use, either 'Occupied heating setpoint' or 'Occupied heating setpoint scheduled' . The two values will sync ( https://github.com/Koenkk/zigbee-herdsman-converters/pull/4734#issuecomment-1313229257). If you physically change the temperature on the TRV it's always more aggressive.  Personally I prefer having the choice to select what I want to do.

I've only ever seen 'Setpoint change source' reported as externally or manual.

 I think the main problem is the basic understanding of when to do remote temperature updates, I posted this recently and that interpretation seems to work https://community.home-assistant.io/t/danfoss-ally-trv-working-with-remote-temp-sensor/276686/192?u=surreallo

**Reply by robertalexa** [reply]
Sorry didn't mean to make it more confusing than it already is. Indeed the behaviour is separate just as you have described and i do not have an issue with it.

My point is only that no "breaking change' is introduced as to what is the default behaviour. E.g. swapping behaviour around, or linking the algorithm factor into the aggressive setpoint changing how it naturally behaves. As that would mess with a lot of people's existing setups.

Apart from that i am happy to assist if needed and test for anyone that will try to improve the behaviour.

**Reply by surrealo** [reply]
yeah I would be very unhappy if suddenly the default behaviour changed :)

**Comment by 0x0013** [comment]
I think this is due to both setpoint attributes reading the setpoint value reported from the TRV. There is only one value on the TRV, it just gets set with a different parameter, depending on which approach is called. So it makes sense that the value gets updated on both - same as it gets updated if you physically turn the valve, even though you didn't set any of the attributes through zigbee.

However, I think this is just the reported value - I don't think that updating of the attribute current reported value actually sends the changed value back via a SET command.

**Comment by Retr0Joe** [comment]
Well,  it seems that some people have different interpretations how this should work. Actually it is not right how it is. It is not okay, when i remotely set the temperature, that "occupied_heating_setpoint" and "occupied_heating_setpoint_scheduled" will be set. They should not be synced. This is wrong. There are maybe more possibilities to change this. But we should discuss this. Only because of "i dont want a breaking change", we shouldnt let it wrong.

When i send occupied_heating_setpoint via automation, it every time uses: "occupied_heating_setpoint_scheduled".

[glad-wolf](https://github.com/glad-wolf) described it it right. We need both possibilities, aggressive and non aggressive.
https://github.com/Koenkk/zigbee-herdsman-converters/pull/4734#issuecomment-1301138813

Look at 0x0201 0x40 Setpoint Command:
Setpoint command sends: setpointType
(enum8) + HeatingSetpoint (16bit)
if setpointType = 1 the actuator will make a
large movement to minimize reaction time to UI.
If setpointType = 0 the behavior will be the
same as setting the attribute "Occupied Heating
Setpoint" to the same value.
if setpointType = 2 displayed setpoint is not
effected but regulated setpoint will change. can
be used for Forecast functionality

So, maybe we need a "switch" to change between "occupied_heating_setpoint" and "occupied_heating_setpoint_scheduled" which could be set at the gui and via a custom command.

There are more wrong things at the basic configuration:
The initial state of "Adaptation run control" should be "none". It does not make sense, that it is set to "initiate_adaption". Because when you mount a new thermostat or reset it, it has no saved valve response. You then should set it to "initiate_adaption" when state is like this: {{ trigger.payload_json.adaptation_run_status == 'none' or trigger.payload_json.adaptation_run_status == 'lost' }}. When state is "found", it should reset to "none" When state is "initiate_adaption" you should be able to set it to "cancel_adaption".

**Reply by surrealo** [reply]
> So, maybe we need a "switch" to change between "occupied_heating_setpoint" and "occupied_heating_setpoint_scheduled" which could be set at the gui and via a custom command.

maybe, however if it needs to be sent within the same payload for it to work I don't know how that would be done since it doesn't seem to be a stored value. occupied_heating_setpoint and occupied_heating_setpoint_scheduled is essentially that 'switch'.

> The initial state of "Adaptation run control" should be "none". It does not make sense, that it is set to "initiate_adaption".

it is set to none,  I just reset 8 TRVs (after firmware upgrade because of the firmware version bug) and all of them are set to none.

IMHO zigbe2qtt should just report and set the values and not have any logic. If you want logic that's what automations are for. I don't know if the TRVs report back the value of the Adaptation run control. If they do then maybe that's something that could be added.

**Reply by robertalexa** [reply]
Ok let me make one final intervention and then i will stay out of the conversation, as it seems you took my comment very personal.

I invite to have a look at all the dafoss ally, popp trv and hive trv in z2m and understand how back i go with this device. And i did simply from frustration of initial incomplete implementations.

A few years ago the default has a slow valve actuation and almost none of the current features implemented. The devices reported temperature and thats pretty much it. I, with the help of @sjorge, @koen and various other members of the community expanded the featureset. By a significant amount. The first and most important thing i changef was the valve to aggressive, just like it happens when you manually set the temp on yhe device. That should be the default. Always, no matter what people want to think of it. Input sth output sth. Anything in between should not be the responsibility of the controller, but rather the automation logic.

Later on there was a PR introducing an extra property for people that prefer to use the slow valve. This is how this second thing came into existance. It is a replica of the default behaviour, with one bit different. I have left the same feedback there. If you experience overshooting it means you failed you learn how the device works with your radiator, your boiler, your house, the external temp, and failed to account for that in your rules for setting the temperature. Of course this is oversimplified, and i am trully open to learning more about this and maybe even change my own automations to make use of it. Since then, nobody came back to me comments with any feedback.

I have not investigated the issue you have described, so i do not have any remarks over are you right or wrong. My only remark is as follows. However this PR improves the controller, it should not change the current default behaviour. An aggressive valve turn just like manually controlling the valve. The reasoning is that if i want to automate, i want to be able to have predictable and replicable behaviour. The slow valve turning doesnt offer that. Proven by my findings as well as tens of other people saying the same all over the net.

Ps. Do not take this personally, i never intended to write an essay and go 2 - 3 years into the past. All the best

**Reply by surrealo** [reply]
I for one am definitely very grateful for all the time and effort put into development :). thanks robertalexa and all the others involved! :)

**Comment by Retr0Joe** [comment]
Hello @robertalexa. I think there are some misinterpretations in here. Let me explain. First of all, i like the work and effort that people who own these devices spent into this project. I own 20 of the devices since november 2022. I started to buy popp ones and sold them later to replace them with the danfoss ones. The reason was, that i wanted to use these also with the danfoss ally app, which is not possible with the rebranded ones. This time i also started with home assistant and zha. Later i changed the coordinator and switched to zigbee2mqtt. At the beginning, i was also very unhappy how these devices react to setpoint changes. Then i tried "better_thermostat", but that was also a fail. Then i found the zigbee specifications and the feature catalgue of these valves and tried to read, understand and change options within zigbee2mqtt.

Believe me, that i tested a lot with these devices and my environment to let them work how they should and i know how gas boilers work. Maybe my first post was not clear enough, maybe i just missed something, that is not well documented. Let my try to explain again.

There are two ways to set the temperature:

aggressive mode:
When temperature is set locally at the thermostat, "occupied heating setpoint" will be locally set and readed from zigbee.

When temperature is set remotely "occupied heating setpoint" **and** "which function" are set?

non aggressive mode:
When temperature is set remotely "occupied heating setpoint scheduled" **and** "which function" are set?

Actually, every time when i change the temperature remotely via home assistant, i see "occupied heating setpoint scheduled". This leads to very slow temperature response/valve opening. I cannot find any function, where i can switch between these modes. It does not make any difference if i send, "climate set temperature" or "occupied heating setpoint" to xx degrees. Every time "occupied heating setpoint scheduled" will be sent, and shown in the logs.

Second thing is, if these values are synced and only ""occupied heating setpoint schedule" is shown at the logs, how can people know which mode is working? As is said, maybe i missed something. Where or what is the option, to change between these two modes?

And my 2 cents is also, that the valve reaction by remotely set temperature should be the same, as when it is locally set at the thermostat. But actually it is not at my environment.

**Reply by robertalexa** [reply]
Ok. In that case, that is worrysome. To my understanding, the regular setpoint should have been bound to the HA thermostat behaviour. While the scheduled setpoint would have been something that people manually called if they preferred that behaviour.

Your comment suggests that is not the case. And if it is not, then that is indeed a big problem. I am currently undergoing renovations in the house, but will do my best to check things at my end asap, as if ehat you are saying is true, then everyones trvs are slow by default.

Thank you kindly for your comment and clarifications, ive never intended to step on your toes.

Will let you know what i find asap. 

Ps. I think in principle it is possible to see the raw command if herdsman is in debug mode, but i dont remember for sure

**Reply by Retr0Joe** [reply]
Hello robert, thank you for your fast reply. I also did not to intend, to step onto someones toes. I just like to understand how things work like they are specified from a manufacturer. Sometimes, things are not clearly defined and let room for interpretations. Some of the modes and functions of these danfoss device are not very well documented by danfoss. These devices are very powerful, if they will be used correctly. I wonder, why danfoss does not look from time to time to official forums, github or something like this. If they would support the community more, that would lead to more recommendations from peoples to buy these products.

An example about boiler configurations. Actually, we have two options to communicate with boilers (heat pumps etc.) and our danfoss valves.

1st) You own a gas boiler. The gas boilers has a physical connection and you own at once Danfoss Ally Boiler-Relay  which is physically attached to your boiler and it can be set on and of with these relay (or another external switch, which you can set via zigbee or wifi). The valves have the states "heat required" and "idle". You can now make an automation, lets say, if 3 of 15 radiators call for heat required, then set the relay and the boiler can heat.

2nd) This is the one that i use. You own a gas boiler. Depending on the thermal needs of your house and the minimum power that the heater can give, it goes on and off. Because when it gets warm outside, the thermal needs will be lower and the heater begins to cycle. I now have an automation, which sends "heat_available" true to all valves if the heater is on and false if the heater is off. I am not sure, if that has influence on the PID algorithm of the danfoss valve.

3rd) You have external heat and no influence to set the heater on and off. Default is "heat_available" true.

There are many things that could be done with these valves, i really like them. I try to understand now, how i can send not  yet implemented commands directly to the valves and document these via trial and error.

[AI459141322073en-000102.pdf](https://github.com/Koenkk/zigbee2mqtt/files/14001378/AI459141322073en-000102.pdf)

**Comment by Retr0Joe** [comment]
![5](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/49c2b2cb-256b-4eb3-849a-cee218801676)

Whatever or however i set temperature, this is the outcome.

**Reply by robertalexa** [reply]
As this is exposed in HA that is expected behaviour. Both setpoints are expoaed to HA by default. The aggresive entity should be connected to the climate entity though while the slow valve ia the one in your screenshot. Whichever value you change, will change both by design. If you do not use the slow valve, you can disable it in HA. If you do not use the aggressive one, you might be able to replace it with the slow. I havent personally tried so i do not know if the best approach would be via z2m or ha.

With regards to your other comment, i went down that route too, there are a lot of features that the trv can do, or at least provide to you so you can make decisions. I personally didn't make use of a lot of things. In my personal setup i have hive trv (same device as danfoss ally but with custom incomplete huve firmware) and the hive boiler controller with a thermostat. I then rely on the thermostat to dictate the main temp in the middle of the house and make other decisions on a per room basis by temperature only. I have not yet configured any "boost" behaviour where the main thermostat increases a bit because the rads need more. In fact, this is exactly why i stopped using hive software and did it my way in z2m.

I found out the hard way that you end up in pretty much a permanent boost, because rooms will start cooling down. The more rooms, the more chances rooms will demand heating.

I will have a look at z2m and ha logs as soon as i can

**Comment by surrealo** [comment]
> Actually, every time when i change the temperature remotely via home assistant, i see "occupied heating setpoint scheduled". This leads to very slow temperature response/valve opening. I cannot find any function, where i can switch between these modes. It does not make any difference if i send, "climate set temperature" or "occupied heating setpoint" to xx degrees. Every time "occupied heating setpoint scheduled" will be sent, and shown in the logs.


that's strange,  I just changed the setpoint via the HA climate card (or whatever it's called) and I still just see externally

![image](https://github.com/Koenkk/zigbee2mqtt/assets/102513879/387c72ed-6a8e-4356-9bf6-af7c95694ccd)

maybe its because I have this set?

![image](https://github.com/Koenkk/zigbee2mqtt/assets/102513879/5c6efd38-a933-4ea0-8afd-250326178499)

out of curiosity I checked the zigbee2mqtt logs regardless of how I change, through HA or the zigbee2mqtt it seems both values get sent over in the payload (  "occupied_heating_setpoint": 19.5,  "occupied_heating_setpoint_scheduled": 19.5, ). 

hehehe now I'm confused again ;) 

@robertalexa , if you want some logs or tests or something, let me know

**Reply by Retr0Joe** [reply]
I also have tried to set "Programming operation mode" to setpoint. The problem is, after you have set it, you cannot unset it. I then have stopped zigbee2mqtt and deleted the value and restarted z2m. There is no difference.

      - service: mqtt.publish
        data_template:
          topic: zigbee2mqtt/{{ "[TCV] Buero" }}/set/programming_operation_mode
          payload: setpoint

      - service: mqtt.publish
        data_template:
          topic: zigbee2mqtt/{{ "[TCV] Buero" }}/set/programming_operation_mode
          payload: ""

**Reply by surrealo** [reply]
ok, I just tried the two 'Programming operation mode' when setpoint is selected the change source will show up as externally when schedule is selected then the change source show's schedule.. I have no idea how that affects anything to be honest. i just got things stable so I'm not going to fool around too much. I always assumed that setting had something to do with the TRV's internal schedule which can be programmed but is not directly exposed in the z2m addon.

**Reply by surrealo** [reply]
this might be a silly question, are you on the latest firmware? does the version show up in the OTA tab or a string of chars like AABBBCCCD? in which case you might want to check out this thread https://github.com/Koenkk/zigbee2mqtt/issues/11049

**Reply by Retr0Joe** [reply]
Yes, i am at the latest firmware. I updated 15 thermostats to version "00.28.0008 00.28". Problem was also before updating.

**Reply by surrealo** [reply]
i suppose you meat "01.28.0008 01.28" :)

**Reply by Retr0Joe** [reply]
That depends on the hardware-version of your thermostats, i have two versions, both are nearly the same. They just actualized some hardware chip.

![7](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/212f0e4c-7925-43d0-ab20-edd81362592c)

**Reply by Retr0Joe** [reply]
Now i tried to set the entity off, but seems not to be allowed via service-call.

description: ""
trigger: []
condition: []
action:
  - service: homeassistant.turn_off
    metadata: {}
    data: {}
    target:
      entity_id: number.tcv_buero_occupied_heating_setpoint_scheduled

![9](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/bd6c64db-d2ba-4aa6-aecf-41b0d6041e0c)

**Comment by Retr0Joe** [comment]
Okay, i think, i found the problem. It seems, that i do not have the entities "occupied_heating_setpoint". I dont know why, but all of my thermostats only have the entity "occupied_heating_setpoint_scheduled". Now it is clear, that only this can be called. The question is, why is it like this? I am very sure, when i bought my first popp thermostats, that i had the entity "occupied_heating_setpoint".

Very strange. Deleting and repairing them does not change anything.

![8](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/9e125633-6967-4dae-835a-936b54223de9)

**Comment by Retr0Joe** [comment]
Now i stopped z2m and renamed all entities at "core.entity_registry" and reboot home assistant os. We will see if that helps.

Edit:
It does not work. Something must have changed. The entities do not exist. I renamed all from "occupied_heating_setpoint_scheduled" to occupied_heating_setpoint and also the descriptions. See below.

![10](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/6dfd9e2f-fa65-49e5-bab6-73c61368ef78)

**Reply by robertalexa** [reply]
Renaming the entity in ha is wrong. That changes the id in ha, but not the source which is mqtt topic comming from z2m.

Are you sure the entity is not in disabled state in ha? Have a look at the device and confirm that first.
![17058560654051785191264577429641](https://github.com/Koenkk/zigbee2mqtt/assets/9019306/afad8543-06cc-4161-8a7c-09ac6e3ff7bc)

**Reply by Retr0Joe** [reply]
Thank you for your reply. The entity was not there, i checked it. But now the "occupied heating setpoint scheduled" is deactivated.

![11](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/b5163e54-dd9b-4918-9572-f8688fa9e56f)

![12](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/3206259c-6143-46e2-809c-0b34cc712edf)

**Reply by robertalexa** [reply]
I suspect you may have had a configuration issue when the device was first adopted? My advice would be for the sake of testing, to remove the device from z2m, also removing it from ha. To play it safe, restart z2m and ha after that. It shouldnt be necessary by why not. Then in ha go to Entities and see if you have any with errors. A message along the lines of ... This entity is not supported by the integrarion anymore. If it is not needed delete it. Cleanse whatever you have there that might already occupy the same ids as the device you want to pair. Once done pair the trv via z2m and let it go into ha naturally. Check the entities.  Hope this helps

**Reply by Retr0Joe** [reply]
Hm no, everything went flawless when installed them. I already did reapiring like this for tcv buero. Did not change anything.
I renamed the entity at the mqtt device page, i think that should work?

![13](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/daeaa1c6-57c9-4622-a525-4426d6071d93)


You are right, entity name at core.entity_registry is still "occupied heating setpoint scheduled". This is really annoying. :-D
Shoudnt it be possible, to rename in there now? 

I mean this:

      {
        "aliases": [],
        "area_id": null,
        "capabilities": null,
        "config_entry_id": "xxxxxx",
        "device_class": null,
        "device_id": "xxxxxx",
        "disabled_by": "integration",
        "entity_category": null,
        "entity_id": "sensor.tcv_flur_unten_occupied_heating_setpoint_scheduled",
        "hidden_by": null,
        "icon": null,
        "id": "xxxxxx",
        "has_entity_name": true,
        "name": null,
        "options": {
          "conversation": {
            "should_expose": false
          }
        },
        "original_device_class": null,
        "original_icon": null,
        "original_name": "Occupied heating setpoint scheduled",
        "platform": "mqtt",
        "supported_features": 0,
        "translation_key": null,
        "unique_id": "0xyyyyyyyyyyyyyyyy_occupied_heating_setpoint_scheduled_zigbee2mqtt",
        "previous_unique_id": null,
        "unit_of_measurement": "°C"


I have 2 entities at core.restore_state now. It seems to work with renaming.

       {
      "state": {
        "entity_id": "number.tcv_buero_occupied_heating_setpoint",
        "state": "20",
        "attributes": {
          "min": 5.0,
          "max": 32.0,
          "step": 0.5,
          "mode": "auto",
          "unit_of_measurement": "°C",
          "friendly_name": "[TCV] Buero Occupied heating setpoint"
        },
        "last_changed": "2024-01-21T16:59:48.198126+00:00",
        "last_updated": "2024-01-21T16:59:48.198126+00:00",
        "context": {
          "id": "01HMPFDK364P4VRVY5VZY067PV",
          "parent_id": null,
          "user_id": null
        }
      },
      "extra_data": {
        "native_max_value": 32.0,
        "native_min_value": 5.0,
        "native_step": 0.5,
        "native_unit_of_measurement": "°C",
        "native_value": 20
      },
      "last_seen": "2024-01-21T17:25:01.597031+00:00"
    },


    {
      "state": {
        "entity_id": "number.tcv_buero_occupied_heating_setpoint_scheduled",
        "state": "20",
        "attributes": {
          "min": 5.0,
          "max": 32.0,
          "step": 0.5,
          "mode": "auto",
          "unit_of_measurement": "°C",
          "friendly_name": "[TCV] Buero Occupied heating setpoint scheduled"
        },
        "last_changed": "2024-01-21T16:07:50.355237+00:00",
        "last_updated": "2024-01-21T16:07:50.355237+00:00",
        "context": {
          "id": "01HMPCEEAK0ZDE1DH7KQXYCQ57",
          "parent_id": null,
          "user_id": null
        }
      },
      "extra_data": {
        "native_max_value": 32.0,
        "native_min_value": 5.0,
        "native_step": 0.5,
        "native_unit_of_measurement": "°C",
        "native_value": 20
      },
      "last_seen": "2024-01-21T16:31:06.386269+00:00"

**Reply by surrealo** [reply]
so finally,  are we supposed to have both entities? occupied heating setpoint and occupied heating setpoint scheduled? I only have the scheduled ones

**Reply by Retr0Joe** [reply]
Well, if you also have only the scheduled entities, we have a problem which should be investigated by someone. I think, then many more people have this issue and just dont know about that.

It seems, that renaming at the mqtt page did it. Just try it with 1 valve, you have nothing to lose. Then make temperature change and check valve response. Be sure, that then 1 of the entities is deactivated. For me that that was automatically the scheduled one.

**Reply by surrealo** [reply]
hmm didn't seem to work for me.. I still only saw the scheduled entity but deactivated. I'm not going to mess with it too much since scheduled is what I want actually and I'm not proficient enough in HA/z2m to be able to recover gracefully if I mess something up.

**Reply by Retr0Joe** [reply]
If it is okay for you, then let it how it is. We should just keep in mind, that it could lead to bad experience for other poeple.

**Reply by surrealo** [reply]
yes, for sure

**Reply by robertalexa** [reply]
You will not see the aggressive entity in HA. I have said that already. You keep saying contradictory information. That function is already connected to the climate entity of the trv in HA. When you set the temperature from the Gauge in HA, that is the aggressive setpoint (desired temp) and current temp working together. They are simply connected behind the scenes into what HA expects.

The scheduled entity is a Separate entity, disabled by default, that people need to enable and use in their automations, should they prefer that behaviour. It is not the duty of the controller to provide mechanisms for toggling between the two. This is something the user should do, through whatever means they find necessary. Remember that z2m is not a tool designed specifically for HA, but an independent zigbee software and hardware controller. The fact it is integrated with HA is just a nice bonus.

If you want to use the schedule setpoint, aka the slow one, you will then need to create yourself a generic thermostat device in HA and connect to entities to it, and not use the default climate entity provided by z2m via mqtt. Remember to enable the entities first.

At this point i no longer believe there is an issue, but rather a misunderstanding of how these pieces connect with each other.

**Reply by Retr0Joe** [reply]
Well, renaming did not make a big difference, i checked with 3 valves now. Slow valve response by changing the target temperature. It should immediately open the valves. I thought, changing the name did it. But when looked into the linked ID´s at the registry, i have seen that you are right. Renaming alone doesnt do anything.

I think someone with more knowledge about how these entities are programmed and linked, should take a look into this. I restored the changed files to old state. I only have the "scheduled" entities again. There are no deactivated "without scheduled" entities to find. surrealo also had also the "scheduled" entities enabled by default. This is no coincidence. :D I dont to use want to use the "slow" response aka "occupied_heating_setpoint_scheduled". This is enabled by default actually. This is the entity which will generated when you pair a new thermostat. There is no "occupied_heating_setpoint" as default, also not hidden or deactivated. :-)

Just try it yourself. Just add a new device, or delete and repair. Then the magic is there. Only the slow entity..

Edit: I think now i understand. I disabled "occupied_heating_setpoint_scheduled" at all the valves now. It is enabled by default. So, when it is disabled, the normal "fast reacting" mode is active. That should be documented somewhere and not be enabled by default. You will also not have any temperature logs at the system log, if these entities are disabled, just checked it with 15 valves now. But you can hear them move, when change target temperature.

![14](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/3cdcf237-d671-4951-ade5-f195cc42d6fa)

![15](https://github.com/Koenkk/zigbee2mqtt/assets/157210636/78706f23-56ea-45c7-827d-8264b5f84dd2)

**Reply by Retr0Joe** [reply]
Temperatue changes with deactivated "occupied_heating_setpoint_scheduled"-entity seem to work. But temperature-changes (set target temperature) will not be shown at the ha-logs anymore. If now the slow or fast modes are activated, no clue. I can live with that, if this is the only issue. Maybe you test yourself when you have time for it. Thanks for the support!

**Reply by robertalexa** [reply]
The clue is simple. Set your valve to 8 degrees. If you hear a lot of noise, that is aggressive. If you hear one single click its slow. Then do thr opposite and set them to 30.

Stop looking for aggressive setpoint. It is already attached to the climate entity. 
https://github.com/Koenkk/zigbee-herdsman-converters/blob/f7cb1f160f07fcca7a354429cad9d63c7272ea85/src/devices/danfoss.ts#L60

![image](https://github.com/Koenkk/zigbee2mqtt/assets/9019306/afd417f3-2c5b-45af-a014-fa3229df7f2b)

And because it is bound to climate, and the majority of people will use yhe default climate, that is the default behaviour.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/9019306/fe850e59-4a85-4534-88ff-40dc002c9a07)

The scheduled entity is extra, not bound to climate, for advanced users that want to make their own climate, and enabled by default. But not being sent to the device unless done by the user. However. It receives updates from z2m to match the climate temperature identically.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/f7cb1f160f07fcca7a354429cad9d63c7272ea85/src/devices/danfoss.ts#L62

![image](https://github.com/Koenkk/zigbee2mqtt/assets/9019306/2874fc68-e748-41af-a525-3645a5a8aac1)

With all of this clarified, and now that i looked at the current code, none if this is an issue. Everything works as per intended implementation.

**Reply by Retr0Joe** [reply]
Thank you for the clarification. The question is now. Must the "occupied_heating_setpoint_scheduled"-entity be disabled or not? I did not really get it, sorry.

From what i see now, it doesnt matter if the "occupied_heating_setpoint_scheduled"entity is enabled? You have to actively set a topic like this for slow response?

service: mqtt.publish
data_template:
  topic: zigbee2mqtt/{{ "[TCV] Buero" }}/set/occupied_heating_setpoint_scheduled
  payload: 20

**Reply by robertalexa** [reply]
If you intend to use the aggresive behaviour, disable it (although it does no harm having it enabled. in my case, i disabled all of the entities i do not use to save on database writes in HA).

![image](https://github.com/Koenkk/zigbee2mqtt/assets/9019306/712c23d1-75ee-4026-8dd9-7cd28d882199)

If you intend to use the slow behaviour, look into [Generic Thermostats](https://www.home-assistant.io/integrations/generic_thermostat/). creat yourself a custom one consisting of the entities required. Then you can disable the climate one provided on the device. You do not need to do service mqtt.publish. You simply set the value on `occupied_heating_setpoint_scheduled` and everyhting is handled by HA -> MQTT -> Z2M

**Reply by Retr0Joe** [reply]
Thank you! Problem solved.

**Reply by robertalexa** [reply]
You are welcome :)

**Comment by ooglyhLL** [comment]
> 1st) You own a gas boiler. The gas boilers has a physical connection and you own at once Danfoss Ally Boiler-Relay which is physically attached to your boiler and it can be set on and of with these relay (or another external switch, which you can set via zigbee or wifi). The valves have the states "heat required" and "idle". You can now make an automation, lets say, if 3 of 15 radiators call for heat required, then set the relay and the boiler can heat.

This is what I had implemented with my previous TRV, based on valve position for lack of other parameters. Worked reasonably well. With my new Popp TRVs, I am a bit clueless what “heat required” relates to. All rooms above setpoint, all valves at 0, or 1%, yet “heat required” flips to 1 for no apparent reason. Any ideas?

**Comment by Miclot-mth** [comment]
It seems that the "heat required" state is related to the "Heat available" command and the "Heating System Mode" parameter (defaults to auto mode).
But I don't understand the interaction of all this.

![image](https://github.com/user-attachments/assets/fbf0fe58-ad3a-44a0-9043-a1de5e1ea840)
